tests/components/vera/common.py

```diff
@@ -0,0 +1,839 @@
+            status_variable_name = None
+
+            if service_id == "urn:upnp-org:serviceId:SwitchPower1":
+                if action == "SetTarget":
+                    status_variable_name = "status"
+            elif service_id == "urn:upnp-org:serviceId:Dimming1":
+                if action == "SetLoadLevelTarget":
+                    status_variable_name = "level"
+            elif service_id == "urn:micasaverde-com:serviceId:SecuritySensor1":
+                if action == "SetArmed":
+                    status_variable_name = "armed"
+            elif service_id == "urn:upnp-org:serviceId:WindowCovering1":
+                if action == "SetLoadLevelTarget":
+                    status_variable_name = "level"
+            elif service_id == "urn:micasaverde-com:serviceId:DoorLock1":
+                if action == "NewTarget":
+                    status_variable_name = "locked"
+            elif service_id == "urn:upnp-org:serviceId:HVAC_UserOperatingMode1":
+                if action == "SetModeTarget":
+                    status_variable_name = "mode"
+            elif service_id == "urn:upnp-org:serviceId:HVAC_FanOperatingMode1":
+                if action == "SetMode":
+                    status_variable_name = "fanmode"
+            elif service_id == "urn:upnp-org:serviceId:TemperatureSetpoint1_Cool":
+                pass
+            elif service_id == "urn:upnp-org:serviceId:TemperatureSetpoint1_Heat":
+                pass
+            elif service_id == "urn:upnp-org:serviceId:TemperatureSetpoint1":
+                if action == "SetCurrentSetpoint":
+                    status_variable_name = "setpoint"
+            elif service_id == "urn:micasaverde-com:serviceId:Color1":
+                if action == "SetColorRGB":
+                    status_variable_name = "CurrentColor"
+
+            device = get_device(device_id, component_data)
+            status = get_device_status(device_id, component_data)
+
+            # Update the device and status objects.
+            if status_variable_name is not None:
+                device[status_variable_name] = state_variable_value
+                status[status_variable_name] = state_variable_value
+
+            # Update the state object.
+            status["states"] = [
+                state
+                for state in status.get("states", [])
+                if state.get("service") != service_id
+                or state.get("variable") != state_variable_name
+            ]
+            status["states"].append(
+                {
+                    "service": service_id,
+                    "variable": state_variable_name,
+                    "value": state_variable_value,
+                }
+            )
+
+            return ResponseStub(json={})
+
+        return ResponseStub(json={})
+
+    def init_controller(base_url: str) -> list:
+        nonlocal data_request
+        controller = VeraController(base_url)
```

Comment:
Patch the controller instead. Don't create a real controller and don't start the thread.

We should be able to manually call the callbacks.

Reply:
Mocking the controller and not starting the thread tests effectively tests a module with different behavior than what is happening when running for real. To ensure proper compatibility, the tests should behave as closely to production as possible. 
Granted, the module shouldn't be using a thread but it is and the tests should accommodate it. Once we have a testing baseline, we can begin to adjust the implementation to avoid it's own thread.

Reply:
It won't matter for home assistant as long as the interface between home assistant and pyvera behaves the same. That should be the goal of these tests. That home assistant interfaces correctly with pyvera and does the correct thing with the result from that interface.

Testing the pyvera behavior is the responsibility of pyvera.